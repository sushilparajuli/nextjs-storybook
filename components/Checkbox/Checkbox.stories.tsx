import type { Meta, StoryObj } from "@storybook/react";
import { expect } from "@storybook/jest";
import { userEvent, within } from "@storybook/testing-library";

import { CheckBox } from "./Checkbox";

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta: Meta<typeof CheckBox> = {
  title: "Components/CheckBox",
  component: CheckBox,
  tags: ["autodocs"],
};

export default meta;
type Story = StoryObj<typeof CheckBox>;

export const BasicCheckbox: Story = {
  play: async ({ args,canvasElement }) => {
    const canvas = within(canvasElement);
    const checkbox = canvas.getByText("âœ”")
    await expect(checkbox).toBeInTheDocument();
    userEvent.click(checkbox);
    await expect(args.onChange).toHaveBeenCalled();
  }
};
