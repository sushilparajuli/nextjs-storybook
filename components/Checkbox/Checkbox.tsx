import styled from "@emotion/styled"
import { ChangeEvent, FC, useId } from "react"
import { Icon } from "../Icon";
import { boxShadow, transition } from "../styles";

type Props = {
    /** onChange callback */
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

const Wrapper = styled.label`
  font-size: 1.8rem;
  & input {
    display: none;
  }
  & input:checked {
    & ~ label {
      ${({ theme }) =>
        boxShadow(theme.components.shadow1, theme.components.shadow2, true)}
      color: ${({ theme }) => theme.font.regular};
    }
  }
`
const Label = styled.label`
display: inline-block;
user-select: none;
cursor: pointer;
text-align: center;
border-radius: 0.5rem;
width: 2.5rem;
height: 2.5rem;
color: rgba(0, 0, 0, 0);
background: ${({ theme }) => theme.components.background};
${({ theme }) =>
  boxShadow(theme.components.shadow1, theme.components.shadow2)}
${transition()};
&:hover {
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2, true)}
}
`;
export const CheckBox : FC<Props> = ({onChange}) => {
  const fieldId = useId()
  return (
    <Wrapper>
      <input id={fieldId} type="checkbox" onChange={onChange} />
      <Label htmlFor={fieldId}>âœ”</Label>
    </Wrapper>
  )
}