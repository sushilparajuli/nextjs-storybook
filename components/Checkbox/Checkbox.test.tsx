import { render, screen } from "@/test-utils";
import userEvent from "@testing-library/user-event";
import { CheckBox } from "./Checkbox";
describe("CheckBox test cases", () => {
  const onChange = jest.fn();

  it("should render", () => {
    const { asFragment } = render(<CheckBox onChange={onChange} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("check onChange  callback", () => {
    render(<CheckBox onChange={onChange} />);
    const element = screen.getByText("âœ”");
    userEvent.click(element);
    expect(onChange).toHaveBeenCalled();
  });
});
