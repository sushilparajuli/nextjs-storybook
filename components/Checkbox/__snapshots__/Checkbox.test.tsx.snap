// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases should render 1`] = `
<DocumentFragment>
  <label
    class="css-f0o7h5"
  >
    <input
      id=":r0:"
      type="checkbox"
    />
    <label
      class="css-lsa8i0"
      for=":r0:"
    >
      âœ”
    </label>
  </label>
</DocumentFragment>
`;
